test/e2e/framework/framework.go

```diff
@@ -328,3 +330,81 @@ func GetAllocation(f *agonesv1.Fleet) *allocationv1.GameServerAllocation {
 			Required: metav1.LabelSelector{MatchLabels: map[string]string{agonesv1.FleetNameLabel: f.ObjectMeta.Name}},
 		}}
 }
+
+// CreateNamespace creates a namespace in the test cluster
+func (f *Framework) CreateNamespace(t *testing.T, namespace string) {
+	t.Helper()
+
+	kubeCore := f.KubeClient.CoreV1()
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
```

Comment:
Do we want to put a label on the namespace?

Reason being, then it would be easy to find all the test namespaces and make the cleanup operation to look for any test namespaces and clear them out -- just in case a defer delete operation doesn't complete because of an error, etc.

WDYT?

Reply:
Good idea. I'll add labels to namespaces.

Reply:
Done